pkg/cluster/update.go

```diff
@@ -190,19 +213,60 @@ func (u *simpleUpgrader) updatePlusOne(ctx context.Context, cs *api.OpenShiftMan
 					return err
 				}
 				vmsBefore[*updated.InstanceID] = struct{}{}
+				blob[instanceName(*updated.Name)] = ssHashes[ssNameForVm(&updated)]
+				if err := u.updateBlob(blob); err != nil {
+					return err
+				}
 			}
 		}
 
 		if err := u.delete(ctx, cs, role, *vm.InstanceID, *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName); err != nil {
 			return err
 		}
+		newBlob := make(map[instanceName]hash)
+		for instancename, h := range blob {
+			if instancename == instanceName(*vm.Name) {
+				continue
+			}
+			newBlob[instancename] = h
+		}
+		if err := u.updateBlob(newBlob); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+func filterOldVMs(vms []compute.VirtualMachineScaleSetVM, blob map[instanceName]hash, ssHashes map[scalesetName]hash) []compute.VirtualMachineScaleSetVM {
+	var oldVMs []compute.VirtualMachineScaleSetVM
+	for _, vm := range vms {
+		if !vmIsCurrent(vm, blob, ssHashes) {
+			oldVMs = append(oldVMs, vm)
+		} else {
+			log.Infof("skipping vm %q since it's already updated", *vm.Name)
+		}
+	}
+	return oldVMs
+}
+
+func vmIsCurrent(vm compute.VirtualMachineScaleSetVM, blob map[instanceName]hash, ssHashes map[scalesetName]hash) bool {
+	for instancename, h := range blob {
```

Comment:
it's a map now.  This whole function reduces to `return blob[instanceName(*vm.Name)] == ssHashes[ssNameForVm(&vm)]`